frontend/packages/topology/src/data-transforms/__tests__/data-transformer.spec.ts

```diff
@@ -162,6 +158,30 @@ describe('data transformer ', () => {
     expect(editUrl).toBe('https://github.com/openshift/console');
   });
 
+  it('should return full git url for GitHub URI', () => {
+    const mockGitURI = 'https://github.com/openshift/console';
+    const editUrl = getEditURL(mockGitURI, 'branch1');
+    expect(editUrl).toBe('https://github.com/openshift/console/tree/branch1');
+  });
+
+  it('should return full git url for GitLab URI', () => {
+    const mockGitURI = 'https://gitlab.com/example/reponame';
+    const editUrl = getEditURL(mockGitURI, 'branch1');
+    expect(editUrl).toBe('https://gitlab.com/example/reponame/-/tree/branch1');
+  });
+
+  it('should return full git url for Bitbucket URI', () => {
+    const mockGitURI = 'https://bitbucket.org/username/examplerepo';
+    const editUrl = getEditURL(mockGitURI, 'branch1');
+    expect(editUrl).toBe('https://bitbucket.org/username/examplerepo/src/branch1');
+  });
+
+  it('should return only git url for repo from non-supported git provider', () => {
+    const mockGitURI = 'https://example.com/username/examplerepo';
+    const editUrl = getEditURL(mockGitURI, 'branch1');
+    expect(editUrl).toBe('https://example.com/username/examplerepo');
+  });
+
```

Comment:
Can you please add also cases for custom domains like `github.example.com`, `gitlab.example.com` and `bitbucket.example.com`? In this cases I also expect the full GitHub branch pathname.

I expect we can not determinate `git.example.com`, but that's acceptable for me at the moment.

Reply:
Updated.